staging/src/k8s.io/kubectl/pkg/cmd/kuberc/kuberc.go

```diff
@@ -0,0 +1,199 @@
+		kuberc.Command,
+	}
+}
+
+func (h *DefaultKubercHandler) InjectAliases(rootCmd *cobra.Command, args []string) {
+	// Register all aliases
+	for alias, command := range h.Command.Aliases {
+		commands := strings.Split(command.Command, " ")
+		cmd, _, err := rootCmd.Find(commands)
+		if err != nil {
+			klog.Warningf("Command %q not found to set alias %q", commands, alias)
+			continue
+		}
+
+		// do not allow shadowing built-ins
+		if _, _, err := rootCmd.Find([]string{alias}); err == nil {
+			klog.Warningf("Setting alias %q to a built-in command is not supported", alias)
+			continue
+		}
+
+		// register alias
+		cmd.Aliases = append(cmd.Aliases, alias)
+
+		// inject alias flags if this is the command that is being targetted
+		// fullAliasCmdPath is the command path defined in kuberc minus the
+		// last command (which is being aliased) and the last arg of the cmdArgs
+		// (which is what the alias would be). The cmdArgs will also include
+		// kubectl so we will ignore the first entry in that array
+		//
+		// Example: kuberc defines alias "raw" for "config view" subcommand,
+		// the user thus will supply "kubectl config raw" on the command line
+		// so we take the command defined in the kuberc file and drop the last
+		// subcommand, which is view, and replace it with the alias defined in
+		// the kuberc file, giving us "kubectl config raw", then we check to
+		// see if that is equal to commands supplied by the user.
+		fullAliasCmdPath := append(commands[:len(commands)-1], alias)
+		if reflect.DeepEqual(fullAliasCmdPath, args[1:]) {
+			klog.V(2).Infof("using alias %q, adding flags...", alias)
+			cmd.Flags().Parse(command.Flags)
+		}
+	}
+}
+
+func (h *DefaultKubercHandler) InjectOverrides(rootCmd *cobra.Command, args []string) {
+	cmd, _, err := rootCmd.Find(args)
+	if err != nil {
+		klog.Warningf("could not find command %q", args)
+		return
+	}
+	if flags, found := h.Command.Overrides[strings.Join(args, " ")]; found {
+		klog.V(2).Infof("adding default flags %q for command %q", flags.Flags, strings.Join(args, " "))
+		if err := cmd.Flags().Parse(flags.Flags); err != nil {
+			klog.Warningf("error parsing flags - %q", err)
+		}
+	}
+}
+
+func (h *DefaultKubercHandler) InjectOverridesRoot(flags *genericclioptions.ConfigFlags, args []string) {
+	if kubercFlags, found := h.Command.Overrides["default"]; found {
+		klog.V(2).Infof("adding default flags %q", kubercFlags.Flags)
+		flags.OverwriteDefaultConfigFlags(kubercFlags.Flags, args)
+	}
+}
+
+// LoadKubeRC reads kuberc file and stores the values in a Preferences struct
+// that is then returned
+func LoadKubeRC(path string) (Preferences, error) {
+	kubercBytes, err := os.ReadFile(path)
+	if err != nil {
+		return Preferences{}, err
+	}
+
+	var preferences Preferences
+	// TODO: (mpuckett159) This probably should be UnmarshalStrict but I'm not sure
+	if err := yaml.Unmarshal(kubercBytes, &preferences); err != nil {
+		klog.Warningf("error unmarshalling the yaml for kuberc")
```

Comment:
What happens if user has corrupted `kubeconfig`?, I think it returns error and we should also return error and intercept the request. Otherwise, users will file issues about kuberc not working without knowing that their kuberc yaml is invalid.